rust/datafusion/src/sql/parser.rs

```diff
@@ -77,20 +77,20 @@ pub struct DFParser {
 
 impl DFParser {
     /// Parse the specified tokens
-    pub fn new(sql: &str) -> Result<Self, ParserError> {
-        let dialect = GenericDialect {};
-        let mut tokenizer = Tokenizer::new(&dialect, sql);
+    pub fn new(sql: &str, dialect: &dyn Dialect) -> Result<Self, ParserError> {
```

Comment:
Rather than make this breaking change, let's add a new constructor named `with_dialect` instead.